sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2828,4 +2839,16 @@ class Dataset[T] private[sql](
       Dataset(sparkSession, logicalPlan)
     }
   }
+
+  /** Convert to an RDD of ArrowPayload byte arrays */
+  private[sql] def toArrowPayloadBytes(): RDD[Array[Byte]] = {
+    val schema_captured = this.schema
+    queryExecution.toRdd.mapPartitionsInternal { iter =>
+      val converter = new ArrowConverters
+      val payload = converter.interalRowIterToPayload(iter, schema_captured)
+      val payloadBytes = ArrowConverters.payloadToByteArray(payload, schema_captured)
```

Comment:
Yes, that is a good idea.  It does support having multiple payloads so that would not be too difficult to change, but the only question is how big to make the groups?